### PR TITLE
Use a better and faster method of authentication

### PR DESCRIPTION
**Pull request by BenCat07**
This method can be spammed as much as we want which means we can send it like every 30 seconds instead of having to wait, also means bots can instantly reply to any auth

### CODE REVIEW

File: src/hooks/SendNetMsg.cpp [hunk 1]
```diff
@@ -155,38 +198,46 @@ class AchievementListener : public IGameEventListener2
...
-    EC::Register(
-        EC::Shutdown, []() { g_IEventManager2->RemoveListener(&event_listener); }, "shutdown_event");
+    g_IEventManager2->AddListener(&event_listener, "cl_drawline", false);
     EC::Register(
         EC::LevelInit,
-        []() { // This will make the timer get triggered next test_and_set call
-            identify_timer.last -= std::chrono::minutes(5);
+        []() {
+            // Remove 1 minute from timer so it triggers on connect
+            identify_timer.last -= std::chrono::minutes(1);
```
Comment (TotallyNotElite) [inline src/hooks/SendNetMsg.cpp:236]: This is a terrible hack. Can you don't?

File: src/hooks/SendNetMsg.cpp [hunk 2]
```diff
@@ -37,9 +37,52 @@ void sendAchievementKv(int value)
...
+    int id       = reinterpret_cast<int &>(x_value);
+
+    if (player_idx != 0xDEAD && (id == CAT_IDENTIFY || id == CAT_REPLY))
+    {
+        // Always reply and set on CA7 and only set on CA8
+        bool reply = id == CAT_IDENTIFY;
+        player_info_s info;
+        if (!g_IEngine->GetPlayerInfo(player_idx, &info))
+            return;
+        if (reply && *answerIdentify && player_idx != g_pLocalPlayer->entity_idx && reply_timer.test_and_set(2000))
```
Comment (TotallyNotElite) [inline src/hooks/SendNetMsg.cpp:81]: Edge case: What if an event is sent while this timer has not passed 2 seconds?

Comment (BenCat07) [reply src/hooks/SendNetMsg.cpp:81]: Hm true but we also want to avoid spam

File: src/hooks/SendNetMsg.cpp (hunk 1 above)
Comment (BenCat07) [reply src/hooks/SendNetMsg.cpp:236]: ok

File: src/hooks/SendNetMsg.cpp [hunk 3]
```diff
@@ -155,38 +208,44 @@ class AchievementListener : public IGameEventListener2
...
+            if (!*identify || CE_BAD(LOCAL_E) || !identify_timer.test_and_set(1000 * 30))
                 return;
-            sendIdentifyMessage(false);
+            sendIdentifyMessage(false, false);
         },
         "sendnetmsg_createmove");
     g_IEventManager2->AddListener(&event_listener, "achievement_earned", false);
+    g_IEventManager2->AddListener(&event_listener, "cl_drawline", false);
     EC::Register(
         EC::Shutdown, []() { g_IEventManager2->RemoveListener(&event_listener); }, "shutdown_event");
```
Comment (TotallyNotElite) [inline src/hooks/SendNetMsg.cpp:248]: Does this correctly remove both listeners?

File: src/hooks/SendNetMsg.cpp [hunk 4]
```diff
@@ -155,38 +208,44 @@ class AchievementListener : public IGameEventListener2
...
+                sendIdentifyMessage(true, false);
                 queue_ca8 = false;
             }
-            // 2 minutes between each identify seems ok?
-            if (!*identify || CE_BAD(LOCAL_E) || !identify_timer.test_and_set(1000 * 60 * 2))
+            // Wait 30 seconds between identifies
+            if (!*identify || CE_BAD(LOCAL_E) || !identify_timer.test_and_set(1000 * 30))
                 return;
-            sendIdentifyMessage(false);
+            sendIdentifyMessage(false, false);
```
Comment (TotallyNotElite) [inline src/hooks/SendNetMsg.cpp:242]: So legacy identifies are no longer being sent?

File: src/hooks/SendNetMsg.cpp (hunk 3 above)
Comment (BenCat07) [reply src/hooks/SendNetMsg.cpp:248]: Yes

File: src/hooks/SendNetMsg.cpp (hunk 4 above)
Comment (BenCat07) [reply src/hooks/SendNetMsg.cpp:242]: Yep, only replies

Comment (BenCat07) [reply src/hooks/SendNetMsg.cpp:241]: It will now send one legacy identify on join

**Comment by Codefreak64** [comment]
Why is this still being refined, do you know how much pain us normal players have to go through, bot many bots we have to kick? and for what, to get a cheap laugh? if you want TF2 to DIE! then by all means please continue, but i'm not going to let a whole bunch of salty players try to KILL A GOOD GAME!!

**Comment by TotallyNotElite** [comment]
> if you want TF2 to DIE! then by all means please continue, but i'm not going to let a whole bunch of salty players try to KILL A GOOD GAME!!

If valve actually cared, maybe it wouldn't be dying. Many players gave up on the game a long time ago.

> Why is this still being refined

Cause it's a hobby.

> do you know how much pain us normal players have to go through, bot many bots we have to kick?

Wow, I really don't care. Complain to bot hosters, not devs.

**Pull request merged**